### PR TITLE
FEATURE: Remove broken onebox thumbnail images completely

### PR DESCRIPTION
**Pull request by vinothkannans**
If onebox's thumbnail image is broken or larger then it will remove it completely from the onebox.

### CODE REVIEW

File: lib/cooked_post_processor.rb [hunk 1]
```diff
@@ -361,8 +361,6 @@ def pull_hotlinked_images(bypass_bump = false)
     return if disable_if_low_on_disk_space
     # don't download remote images for posts that are more than n days old
     return unless @post.created_at > (Date.today - SiteSetting.download_remote_images_max_days_old)
-    # we only want to run the job whenever it's changed by a user
-    return if @post.last_editor_id == Discourse.system_user.id
```
Comment (ZogStriP) [inline lib/cooked_post_processor.rb]: Why was this removed?

File: app/jobs/regular/pull_hotlinked_images.rb [hunk 2]
```diff
@@ -107,6 +107,11 @@ def execute(args)
         images = doc.css("img[src]") - doc.css("img.avatar")
         images.each do |tag|
           src = tag['src']
+          if tag['class'] == 'thumbnail' && tag.parent['class'] == 'onebox-body' &&
```
Comment (ZogStriP) [inline app/jobs/regular/pull_hotlinked_images.rb]: Wouldn't it be simpler to do this instead?

```
images = doc.css("img[src]") - doc.css("img.avatar") - doc.css(".onebox-body img.thumbnail")
```

File: lib/cooked_post_processor.rb (hunk 1 above)
Comment (vinothkannans) [reply lib/cooked_post_processor.rb]: After the system_user edit for `downloaded local copies of images` again we need to call `pull_hotlinked_images` to replace broken & large images with placeholder.

In 1st step downloaded local images will replaced in `raw` content. Here post will be updated automatically by system user. Then `CookedPostProcess` will be called and downloaded onebox thumbnail will replaced with local copy. Now we have to call `pull_hotlinked_images` to check broken and large image (we can't do it in 1st step since there raw post only changed. Now we changing cooked content). Now last edit is made by system user. That's why I removed that.

1. `pull_hotlinked_images`
2. `CookedPostProcess`
3. `pull_hotlinked_images`

File: app/jobs/regular/pull_hotlinked_images.rb (hunk 2 above)
Comment (vinothkannans) [reply app/jobs/regular/pull_hotlinked_images.rb]: I don't want to skip those onebox images. I am finding it and removing from cooked html using Nokogiri.

File: lib/cooked_post_processor.rb (hunk 1 above)
Comment (vinothkannans) [reply lib/cooked_post_processor.rb]: @ZogStriP I just removed it from this PR to be it merged. I can add the above code in another PR. But it gives spec error since [this code](https://github.com/discourse/discourse/blob/master/spec/components/cooked_post_processor_spec.rb#L550). Any help?

**Comment by vinothkannans** [comment]
This PR is now mergable :+1:

**Comment by coding-horror** [comment]
Can you review @gschlager?

**Comment by gschlager** [comment]
Looks good to me. Could you please add a test for this? 😉

**Comment by vinothkannans** [comment]
@gschlager RSpec test function added :+1:

**Comment by gschlager** [comment]
Thanks, but I don't think this spec is testing the right thing. It still works even without your changes in `pull_hotlinked_images.rb`. And it tests only for broken images, not for large images.

Could you take another look please?

**Comment by vinothkannans** [comment]
@gschlager I fixed it. And added new spec function for large image also. It looks like travis failed for some other reason.

**Comment by SamSaffron** [comment]
we got to hold off here a bit for @ZogStriP my concern here is the vast majority of the issues we are seeing in the wild is that finaldestination is not following an image redirect correctly.  lets figure that out first

**Comment by SamSaffron** [comment]
Yeah I am closing this for now, we have some serious issues downloading totally valid images that need to be sorted first @gschlager  / @ZogStriP
